main.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+  def my_each
+    pos = 0
+    return to_enum unless block_given?
+
+    loop do
+      break if pos >= length
+
+      yield self[pos]
+      pos += 1
+    end
+  end
+
+  def my_each_with_index
+    pos = 0
+    return to_enum unless block_given?
+
+    loop do
+      break if pos >= length
+
+      yield self[pos], pos
+      pos += 1
+    end
+  end
+
+  def my_select
+    return_arr = []
+    return to_enum unless block_given?
+
+    my_each do |element|
+      return_arr.push(element) if yield element
+    end
+    return_arr
+  end
+
+  def my_all?(arg = nil)
+    return_val = true
+    my_each do |element|
+      return_val = (block_given? && yield(element)) || (arg == element)
+    end
+    return_val
+  end
+
+  def my_any?(arg = nil)
+    return_val = false
+    my_each do |element|
+      return_val = (block_given? && yield(element)) || (arg == element)
+      return true if return_val
+    end
+    return_val
+  end
+
+  def my_none?(arg = nil)
+    return_val = true
+    my_each do |element|
+      return_val = !((block_given? && yield(element)) || (arg == element))
+      return false unless return_val
+    end
+    return_val
+  end
```

Comment:
- For `my_all?`, `my_any?`, `my_none?` handle the following cases
1. when no block or argument is given,
- For the built-in method
`[1, true, 'hi', [ ] ].all?` returns `true`

- Your methods
`[1, true, 'hi', [ ] ].my_all?` returns `false`

2. when the argument is a REGEX or a CLASS.

- For the built-in method
`%w[ant bear cat].all?(/a/) => true`
`[1, 2i, 3.14].all?(Numeric) => true`

- Your methods 
`%w[ant bear cat].my_all?(/a/) => false`
`[1, 2i, 3.14].my_all?(Numeric) => false`